frontend/src/components/viewers/VisualizationCreator.tsx

```diff
@@ -94,10 +102,15 @@ class VisualizationCreator extends Viewer<VisualizationCreatorProps, Visualizati
     // provided, and a visualization type is selected, and a onGenerate function
     // is provided.
     const canGenerate = !isBusy &&
-      !!source.length &&
-      !!selectedType &&
+      (
+        (!!source.length && !!selectedType) ||
```

Comment:
This is hard to read. Consider breaking it into multiple conditions and then combine them. For example:
```
const hasSourceAndSelectedType = source.length > 0 && !!selectedType
const hasCustomTypeWithCode = selectedType === .... && code.length > 0

const canGenerate = !isBusy && (hasSourceAndSelectedType || hasCustomTypeWithCode)
```
Also, be explicit with truthy/falsy statements. `source.length > 0` reads better than `!!source.length` for indicating to the reader that you want a non-zero length of the string in source.